### DISCUSSION TITLE
UI thread in System.Diagnostics.Process.Exited event handler

### DISCUSSION THREAD

**Discussion by lospeq** [microsoft/WindowsAppSDK#1988] [Q&A] [answered]
During the app startup, I'm trying to run an external process and perform some changes in the MainWindow after the process has exited 
```
// calling from the window constructor as an example
// same results calling from App object after full MainWindow initialization and m_window.Activate()

public sealed partial class MainWindow : Window
{
    public MainWindow()
    {
        InitializeComponent();
        RunProcess():
    }

    private void RunProcess() 
    {
        Process install = new Process();
        install.StartInfo.UseShellExecute = true;
        install.StartInfo.Verb = "runas";
        install.StartInfo.FileName = aliasPath;
        install.EnableRaisingEvents = true;
        install.Exited += new EventHandler(install_HasExited);
        install.Start();
    }
    
    private void install_HasExited(object sender, System.EventArgs e)
    {
        // call some main window method
    }
}
```
This, however, throws a 0x8001010E (RPC_E_WRONG_THREAD) error.
All attempts to acquire UI thread with RunAsync or saving dispatchers from the UI thread
```
 m_window.Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () => {});
 CoreApplication.MainView.CoreWindow.Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () => {});
...
```
seem to fail with either "Object reference not set to an instance of an object" or no window is created errors even if called from the window object itself.
Have also tried fiddling with Process properties such as SynchronizingObject to no avail. 

Is there a correct way to handle this?

**Comment by ghost** [comment, marked as answer]
`CoreApplication` / `CoreWindow` / `aka. UWP` is not supported by windowsappsdk. 
Only `Win32` is supported.

**Reply by lospeq** [reply]
Thank you, managed to get it working with MainWindow's DispatcherQueue.TryEnqueue.
